### ISSUE TITLE
After / AfterEach lose the context

### ISSUE THREAD

**Issue by NeverI** [TheLudd/mocha-gwt#2]
Hi again,

I cannot write test without this library anymore, so thanks :+1: 
But here is a little issue:

``` coffeescript
describe 'test', ->
  Given -> @value = 'foo'

  afterEach -> console.log @value  # will be undefined
```

**Comment by TheLudd** [comment]
I have noticed this as well. Will look into it.

**Comment by yeehaa123** [comment]
Hi, I'm running into this issue too. Any workarounds?

**Comment by TheLudd** [comment]
Did some research here and it seems like an implementation detail causes this error. At first glance it will be difficult to fix. 

Just to get things straight... We are talking about `afterAll` here? No `afterEach` is defined.

The `afterAll` is defined in a parent suite wrapping all `describe` blocks in the file. I am guessing that this causes the error. If I put the `afterAll` on the same level as the files main ` describe`  block the error seems to disappear but a whole lot of other errors occur.

**Comment by yeehaa123** [comment]
@TheLudd No, actually, afterEach that does not work properly. Like @NeverI shows in his example, instance variables are undefined in afterEach blocks

Love the library, btw. Thanks for all the great work!
